finetrainers/trainer.py

```diff
@@ -725,30 +780,31 @@ def train(self) -> None:
 
         accelerator.wait_for_everyone()
         if accelerator.is_main_process:
+            # TODO: consider factoring this out when supporting other types of training algos.
             self.transformer = unwrap_model(accelerator, self.transformer)
-            dtype = (
-                torch.float16
-                if self.args.mixed_precision == "fp16"
-                else torch.bfloat16
-                if self.args.mixed_precision == "bf16"
-                else torch.float32
-            )
-            self.transformer = self.transformer.to(dtype)
```

Comment:
`transformer` should already be in the right data-type. If we want the users to upcast before getting the final state dict, that is a different concern.